File: src/main/java/src/baseDatos/Deserializador.java
package baseDatos;

import java.io.FileInputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.IOException;
import java.io.ObjectInput;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.util.ArrayList;
import java.util.List;

import uiMain.Main;
import gestorAplicacion.*;
import gestorAplicacion.Calendario.Beca;
import gestorAplicacion.Calendario.Facultad;
import gestorAplicacion.Calendario.Horario;
import gestorAplicacion.Calendario.Materia;
import gestorAplicacion.Calendario.Tarea;
import gestorAplicacion.Calendario.TareaEstudiante;
import gestorAplicacion.Calendario.gestionDatos;
import gestorAplicacion.personas.Estudiante;
import gestorAplicacion.personas.Persona;
import gestorAplicacion.personas.Profesor;


public class Deserializador{

    public static void deserializar(gestionDatos gestor) {
    	deserializarEstudiantes(gestor,new File("src\\baseDatos\\temp\\Estudiantes.txt"));
    	deserializarProfesores(gestor,new File("src\\baseDatos\\temp\\Profesores.txt"));
    	deserializarMaterias(gestor,new File("src\\baseDatos\\temp\\Materias.txt"));
    	deserializarBeca(gestor,new File("src\\baseDatos\\temp\\Beca.txt"));
    }

    public static void deserializarEstudiantes(gestionDatos gestor, File ruta) {
        try {
            FileInputStream file = new FileInputStream(ruta);
            ObjectInputStream in = new ObjectInputStream(file);
            gestor.setEstudiantes((ArrayList<Estudiante>) in.readObject());

        } catch (IOException | ClassNotFoundException e) {
            System.out.println("Error en la serializacion" + e);
        }
    }

    public static void deserializarProfesores(gestionDatos gestor, File ruta) {
        try {
            FileInputStream file = new FileInputStream(ruta);
            ObjectInputStream in = new ObjectInputStream(file);
            gestor.setProfesores((ArrayList<Profesor>) in.readObject());

        } catch (IOException | ClassNotFoundException e) {
            System.out.println("Error en la serializacion" + e);
        }
    }

    public static void deserializarMaterias(gestionDatos gestor, File ruta) {
        try {
            FileInputStream file = new FileInputStream(ruta);
            ObjectInputStream in = new ObjectInputStream(file);
            gestor.setMaterias((ArrayList<Materia>) in.readObject());

        } catch (IOException | ClassNotFoundException e) {
            System.out.println("Error en la serializacion" + e);
        }
    }

    public static void deserializarBeca(gestionDatos gestor, File ruta) {
        try {
            FileInputStream file = new FileInputStream(ruta);
            ObjectInputStream in = new ObjectInputStream(file);
            gestor.setSistemaBecas((Beca) in.readObject());

        } catch (IOException | ClassNotFoundException e) {
            System.out.println("Error en la serializacion" + e);
        }
    }

}

File: src_py/baseDatos/Deserializador.py
import pickle
import os

class Deserializador:
    @staticmethod
    def deserializar(gestor):
        base_path = os.path.dirname(os.path.abspath(__file__))
        temp_path = os.path.join(base_path, "temp")
        os.makedirs(temp_path, exist_ok=True)  # Create 'temp' directory if it doesn't exist

        est_path = os.path.join(temp_path, "Estudiantes.pkl")
        prof_path = os.path.join(temp_path, "Profesores.pkl")
        mat_path = os.path.join(temp_path, "Materias.pkl")
        beca_path = os.path.join(temp_path, "Beca.pkl")

        Deserializador.deserializarEstudiantes(gestor, est_path)
        Deserializador.deserializarProfesores(gestor, prof_path)
        Deserializador.deserializarMaterias(gestor, mat_path)
        Deserializador.deserializarBeca(gestor, beca_path)
    @staticmethod
    def deserializarEstudiantes(gestor, ruta):
        try:
            with open(ruta, "rb") as file:
                gestor.setEstudiantes(pickle.load(file))
        except FileNotFoundError as e:
            print("Error en la deserialización:", e)

    @staticmethod
    def deserializarProfesores(gestor, ruta):
        try:
            with open(ruta, "rb") as file:
                gestor.setProfesores(pickle.load(file))
        except FileNotFoundError as e:
            print("Error en la deserialización:", e)

    @staticmethod
    def deserializarMaterias(gestor, ruta):
        try:
            with open(ruta, "rb") as file:
                gestor.setMaterias(pickle.load(file))
        except FileNotFoundError as e:
            print("Error en la deserialización:", e)

    @staticmethod
    def deserializarBeca(gestor, ruta):
        try:
            with open(ruta, "rb") as file:
                gestor.setSistemaBecas(pickle.load(file))
        except FileNotFoundError as e:
            print("Error en la deserialización:", e)
